mixnode/src/commands/upgrade.rs

```diff
@@ -301,3 +383,39 @@ pub fn execute(matches: &ArgMatches) {
     // here be upgrade path to 0.9.X and beyond based on version number from config
     do_upgrade(existing_config, matches, package_version)
 }
+
+#[cfg(test)]
+mod upgrade_tests {
+    use super::*;
+    use serial_test::serial;
+
+    #[test]
+    #[serial]
+    fn test_patch_0_10_2_upgrade() {
+        let config = Config::default().with_id("0").with_custom_version("0.10.1");
+        let matches = ArgMatches::default();
+        let old_version = Version::new(0, 10, 1);
+        let new_version = Version::new(0, 10, 2);
+        let new_config =
+            patch_0_10_2_upgrade(config, &matches, &old_version, &new_version).unwrap();
+        assert_eq!(new_config.get_version(), "0.10.2");
+    }
+
+    #[test]
+    #[serial]
+    fn test_patch_0_10_2_upgrade_error() {
```

Comment:
This tests makes me potentially nervous as it saves data where you might already have an initialised node. But I'm also not sure what would be a good alternative for it.

Reply:
I'll replace the id from `0` to `-42`, in the hope that someone running these tests won't think to use that magic number anytime soon :D . A longer term solution would be to have a configurable config directory.